core/display.js

```diff
@@ -10,109 +10,112 @@
 
-    if (!this._target) {
-        throw new Error("Target must be set");
-    }
+        // The visible canvas
+        this._target = target;
 
-    if (typeof this._target === 'string') {
-        throw new Error('target must be a DOM element');
-    }
+        if (!this._target) {
+            throw new Error("Target must be set");
+        }
 
-    if (!this._target.getContext) {
-        throw new Error("no getContext method");
-    }
+        if (typeof this._target === 'string') {
+            throw new Error('target must be a DOM element');
+        }
 
-    this._targetCtx = this._target.getContext('2d');
+        if (!this._target.getContext) {
+            throw new Error("no getContext method");
+        }
 
-    // the visible canvas viewport (i.e. what actually gets seen)
-    this._viewportLoc = { 'x': 0, 'y': 0, 'w': this._target.width, 'h': this._target.height };
+        this._targetCtx = this._target.getContext('2d');
 
-    // The hidden canvas, where we do the actual rendering
-    this._backbuffer = document.createElement('canvas');
-    this._drawCtx = this._backbuffer.getContext('2d');
+        // the visible canvas viewport (i.e. what actually gets seen)
+        this._viewportLoc = { 'x': 0, 'y': 0, 'w': this._target.width, 'h': this._target.height };
 
-    this._damageBounds = { left:0, top:0,
-                           right: this._backbuffer.width,
-                           bottom: this._backbuffer.height };
+        // The hidden canvas, where we do the actual rendering
+        this._backbuffer = document.createElement('canvas');
+        this._drawCtx = this._backbuffer.getContext('2d');
 
-    Log.Debug("User Agent: " + navigator.userAgent);
+        this._damageBounds = { left:0, top:0,
+                               right: this._backbuffer.width,
+                               bottom: this._backbuffer.height };
 
-    this.clear();
+        Log.Debug("User Agent: " + navigator.userAgent);
 
-    // Check canvas features
-    if (!('createImageData' in this._drawCtx)) {
-        throw new Error("Canvas does not support createImageData");
-    }
+        this.clear();
 
-    this._tile16x16 = this._drawCtx.createImageData(16, 16);
-    Log.Debug("<< Display.constructor");
-}
+        // Check canvas features
+        if (!('createImageData' in this._drawCtx)) {
+            throw new Error("Canvas does not support createImageData");
+        }
 
-let SUPPORTS_IMAGEDATA_CONSTRUCTOR = false;
-try {
-    new ImageData(new Uint8ClampedArray(4), 1, 1);
-    SUPPORTS_IMAGEDATA_CONSTRUCTOR = true;
-} catch (ex) {
-    // ignore failure
-}
+        this._tile16x16 = this._drawCtx.createImageData(16, 16);
+        Log.Debug("<< Display.constructor");
+
+        // ===== PROPERTIES =====
+
+        this._scale = 1.0;
+        this._clipViewport = false;
+        this.logo = null;
+
+        // ===== EVENT HANDLERS =====
+
+        this.onflush = () => {}; // A flush request has finished
```

Comment:
I was thinking even further up, so we can avoiding splitting up the two parts of the properties. :)
(this is fine as well though)

Reply:
I'm not sure if I understand. Where were you planning to put these? Merged with the properties?

Reply:
Basically something like this:

```Javascript
        this._tile16x16 = this._drawCtx.createImageData(16, 16);
        Log.Debug("<< Display.constructor");

        // ===== EVENT HANDLERS =====

        this.onflush = () => {}; // A flush request has finished

        // ===== PROPERTIES =====

        this._scale = 1.0;
        this._clipViewport = false;
        this.logo = null;
    }

    // ===== PROPERTIES (cont.) =====

    get scale() { return this._scale; }
    set scale(scale) {
        this._rescale(scale);
    }

```

But those old school event handlers will eventually be removed, so it is perhaps nothing to bother with...